### PR TITLE
Added an API call to fetch exportable TestCase

### PR DESCRIPTION
**Pull request by RohitKandimalla**
## MADiE PR

Jira Ticket: [MAT-0000](https://jira.cms.gov/browse/MAT-0000)
(Optional) Related Tickets:

### Summary

- There is no assigned ticket, but this PR is just to fetch the real test case export from our backend service instead of mocked data. 

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included
* [x] No extemporaneous files are included (i.e Complied files or testing results)
* [x] This PR is into the **correct branch**.
* [ ] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: src/components/testCaseLanding/qiCore/TestCaseList.tsx [hunk 1]
```diff
@@ -227,6 +229,37 @@ const TestCaseList = (props: TestCaseListProps) => {
...
+      );
+      const url = window.URL.createObjectURL(exportData);
+      const link = document.createElement("a");
+      link.href = url;
+      link.setAttribute(
+        "download",
+        `${ecqmTitle}-v${version}-${getModelFamily(model)}-TestCases.zip`
+      );
+      document.body.appendChild(link);
+      link.click();
```
Comment (nmorasb) [inline src/components/testCaseLanding/qiCore/TestCaseList.tsx]: might be able to replace 241-249 with the `file-saver` utility that offers a `saveAs` function for file downloads

Comment (RohitKandimalla) [reply src/components/testCaseLanding/qiCore/TestCaseList.tsx]: Looking into it

**Review by nmorasb — APPROVED** [review]
